spec/models/provider_relationship_spec.rb

```diff
@@ -3,6 +3,13 @@
 require "rails_helper"
 
 RSpec.describe ProviderRelationship, type: :model do
+  let(:lead_provider) { create(:lead_provider) }
+  let(:delivery_partner) { create(:delivery_partner) }
+  let(:cohort) { create(:cohort) }
+  let!(:provider_relationship) do
```

Comment:
Out of curiosity - what is your general rule of thumb between useing before each block and let!

Reply:
I avoid let!  as much as possible these days but still use it. The problem is when you do it in a top scope and it ends up creating other objects in other specs where those objects are not required. In those cases it's probably better to use before.

Reply:
No particular reason for it, but generally when I use the named thing that was created I use `let!`, otherwise a before to just set up state for the test